pyanaconda/network.py

```diff
@@ -94,6 +94,23 @@ def check_ip_address(address, version=None):
         return False
 
 
+def is_valid_hostname_to_set(hostname):
+    """Check if the given string is (syntactically) a valid hostname for the running system.
```

Comment:
I know it's not easy but we should definitely have better description for how `is_valid_hostname_to_set` differs from `is_valid_hostname`. From what is written here it doesn't explain why we have both methods. If this is for running system for what is the second method?

Maybe call this method `is_valid_hostname_without_trailing_dot` and explain in description why we need to have hostname without the trailing dot?

Reply:
I don't think `_without_trailing_dot` is the right way out. The underlying reason is that different rules apply, not just the dot. Related to that is that I see the real hurdle for future comprehension as "hostname" now becoming an ambiguous word. Are we talking about hostname for us, or for anyone else on the network? So I tried to follow that split (us/others).

How about this?
- `is_valid_hostname_for_this_system`
- `is_valid_hostname_of_other_servers`

There are also these possible changes:
- Have two parallel implementations of the check (I really dislike that as it touches too much to my liking... but don't rule it out yet)
- Rename everything (I don't like that but it seems more clear than just hostname)
- Put more information into doc strings of the two methods

Reply:
Maybe, we could just extend the function is_valid_hostname to:
`is_valid_hostname(hostname, allow_trailing_dot=False)`

Reply:
Conclusion from team discussion: Keep one function, add a parameter `static` or `local` that defaults to `False` and activates the more strict check.